src/plugins/data_view_field_editor/public/components/preview/preview_controller.tsx

```diff
@@ -108,6 +139,143 @@ export class PreviewController {
     documentId: undefined,
   };
 
+  getNamesNotAllowed = () => {
+    if (!this.namesNotAllowed) {
+      const fieldNames = this.dataView.fields.map((fld) => fld.name);
+      const runtimeCompositeNames = Object.entries(this.dataView.getAllRuntimeFields())
+        .filter(([, _runtimeField]) => _runtimeField.type === 'composite')
+        .map(([_runtimeFieldName]) => _runtimeFieldName);
+      this.namesNotAllowed = {
+        fields: fieldNames,
+        runtimeComposites: runtimeCompositeNames,
+      };
+    }
+
+    return this.namesNotAllowed;
```

Comment:
Nit: Do we need a private variable here or can this just be simplified by using `once` or `memoize`?

```ts
    getNamesNotAllowed = once(() => {
      const fieldNames = this.dataView.fields.map((fld) => fld.name);
      const runtimeCompositeNames = Object.entries(this.dataView.getAllRuntimeFields())
        .filter(([, _runtimeField]) => _runtimeField.type === 'composite')
        .map(([_runtimeFieldName]) => _runtimeFieldName);
      this.namesNotAllowed = {
        fields: fieldNames,
        runtimeComposites: runtimeCompositeNames,
      };
    });
```

Also, is there any way the data view can change so that the variable stored here might actually be the incorrect list of names? I wasn't able to get it to happen, but I might be missing something.

Reply:
>Nit: Do we need a private variable here or can this just be simplified by using once or memoize?

Good suggestion!

>Also, is there any way the data view can change

Thats not possible. The controller has the same lifecycle as the flyout - in order to work with a different data view another controller needs to be instantiated.